cogs/displayinfo.py

```diff
@@ -33,85 +33,130 @@ class DisplayInfo(commands.Cog):
 
     def __init__(self,bot):
         self.bot = bot
-        self.timers = {}
-        self.current_collection = ""
+        self.current_collection = db["UserTracking"] # TODO: Move this secrets file
+        self.timers = self.current_collection.find_one() # Gets document from collection.
         self.current_guild_name = ""
         self.current_date = str(date.today())
+        self.members_online = set()
+        self.timed_task = None
+    
+    '''
+    Sample dictionary layout: {'_id': ObjectId('64d6c8e896c982ece6550d23'), '326586435958341636': {'2023-08-14': { 'Activity 1': {'totalTime': [], 'startTime': []}, 'Activity 2': {'totalTime': [], 'startTime': []}}}, 
+    '2023-08-15': {'Activity 1': {'totalTime': [], 'startTime': []}
+    '''
+    
+    @commands.Cog.listener()
+    async def on_ready(self):
+        for guild in self.bot.guilds:
+            for member in guild.members:
+                if member.status == discord.Status.online and member.activity:
+                    self.members_online.add(member)
+                    after_string_id = str(member.id)
+                    if after_string_id not in self.timers:
+                        self.timers[after_string_id] = {} # If the member is not already tracked, add them
+                    if self.current_date not in self.timers[after_string_id]:
+                        self.timers[after_string_id][self.current_date] = {} # If the current date is not present in the member's dictionary, add it
+                    if member.activity.name not in self.timers[after_string_id][self.current_date]: # If the member's current activity/game/status is not present, add it with initial values
+                        self.timers[after_string_id][self.current_date][member.activity.name] = {
+                            "totalTime" : 0, 
+                            "startTime" : 0,
+                        }
+                    self.timers[after_string_id][self.current_date][member.activity.name]["startTime"] = datetime.now() # Record the start time of the member's current activity
+
+        print(self.members_online)
+
+        if len(self.members_online):
+            self.timed_task = asyncio.create_task(self.timed_update()) #start timed_update once bot goes online if a member is online
+
+
         
     # The 'mem_before' parameter represents the member's previous state, and 'mem_after' represents the current state.
     @commands.Cog.listener()
     async def on_member_update(self, mem_before:discord.Member, mem_after:discord.Member):
+        self.current_guild_name = mem_before.guild.name
+        
         if mem_before.activity and mem_before.activity == mem_after.activity:
```

Comment:
Will this check work if someone goes offline? Their activity might not have changed but they went offline completely. Might need to look into this.